src/components/NewEventDialog.js

```diff
@@ -0,0 +1,173 @@
+import React, { useCallback } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import useForm from 'react-hook-form';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import { TextField, Button, Grid, Box } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import gql from 'graphql-tag';
+import useStyle from './UseStyle';
+import { DateTime } from 'luxon';
+
+export const mutationAddEvent = gql`
+  mutation addEvent(
+    $title: String!
+    $start: String!
+    $end: String!
+    $description: String!
+  ) {
+    addEvent(
+      title: $title
+      start: $start
+      end: $end
+      description: $description
+    ) {
+      id
+      title
+      start
+      end
+      description
+    }
+  }
+`;
+
+export const NewEventDialog = props => {
+  const { onClose, open, selectedDate } = props;
+  const classes = useStyle();
+
+  const { handleSubmit, register, errors } = useForm();
+  const [addEvent, { loading, error }] = useMutation(mutationAddEvent);
+
+  /* date format: yyyy-mm-dd, startTime/endTime: hh:mm.
+   * convert to string yyyy-mm-ddThh:mmZ
+   * parsing with Luxon (fromISO)  to format that supported by all browsers
+   */
+
+  const onSubmit = useCallback(
+    ({ title, description, date, startTime, endTime }) => {
+      const startEvent = DateTime.fromISO(date + 'T' + startTime).setZone(
+        'UTC'
+      );
+      const endEvent = DateTime.fromISO(date + 'T' + endTime).setZone('UTC');
+      if (endEvent - startEvent <= 0) {
+        const timeErrror = document.getElementById('timeError');
```

Comment:
I think this is considered a bad practice in React since an `id` will not be [reusable](https://www.javascriptstuff.com/use-refs-not-ids/).

I think here we could use `react-hook-form`. First option is to [setError](https://react-hook-form.com/api/#setError), but I think we could even put this as part of validation as seen in this [question](https://github.com/react-hook-form/react-hook-form/issues/227).

Reply:
Thanks for that!
I added `validate: (value) => value < watch('endTime')` like you suggested